include/bluetooth/audio/audio.h

```diff
@@ -1516,6 +1516,45 @@ int bt_audio_unicast_group_create(struct bt_audio_stream *streams,
 				  uint8_t num_stream,
 				  struct bt_audio_unicast_group **unicast_group);
 
+/** @brief Add streams to a unicast group as a unicast client
+ *
+ *  This function can be used to add additional streams to a
+ *  bt_audio_unicast_group.
+ *
+ *  This can be called at any time before when none of the streams in the
+ *  group has not been started (see bt_audio_stream_ops.started()).
+ *  Once all streams in a group has been stopped
+ *  (see bt_audio_stream_ops.stopped()), additional streams can be added.
+ *
+ *  @param unicast_group  Pointer to the unicast group
+ *  @param streams        Array of stream objects being added to the group.
+ *  @param num_stream     Number of streams in @p streams.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_audio_unicast_group_add_streams(struct bt_audio_unicast_group *unicast_group,
+				       struct bt_audio_stream *streams,
```

Comment:
The ISO APIs takes arrays of pointers to allow for a more flexible API, so that the channels do not have to be in an array. Should we update the Audio API to the same?

Reply:
I would need some discussion to have an opinion on that.

Reply:
Some details for the discussion: 

The ISO API uses an array of pointers, such you can e.g. allocate your ISO channels as 
```c
struct bt_iso_chan left;
struct bt_iso_chan right;

cig_param.cis_channels[0] = &left;
cig_param.cis_channels[1] = &right;

int bt_iso_cig_create(&cig_param, &cig);
```

whereas this takes an array of channels, so that you need to do
```c
struct bt_audio_stream streams[2];
struct bt_audio_stream *left = &streams[0];
struct bt_audio_stream *right = &streams[1];

bt_audio_unicast_group_add_streams(group, streams);
```

In other words, the channels need to be allocated in an array, which is less flexible IMO, especially if you need to allocate multiple streams per remove device. The alternative here would be 

```c
struct bt_audio_stream *streams[2];
struct bt_audio_stream left;
struct bt_audio_stream right;

streams[0] = &left;
streams[1] = &right;

bt_audio_unicast_group_add_streams(group, streams, ARRAY_SIZE(streams));
```

Reply:
> But that would be a separate PR, I assume.

Indeed, I'll create an issue for it

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/42587